pkg/jobs/delegate_control_test.go

```diff
@@ -425,32 +455,38 @@ func TestJobControlByType(t *testing.T) {
 							_, err := registry.CreateAdoptableJobWithTxn(context.Background(), record, jobID, nil /* txn */)
 							require.NoError(t, err)
 							th.sqlDB.Exec(t, "UPDATE system.jobs SET status=$1 WHERE id=$2", status, jobID)
+							t.Logf("created job %d (%T) with status %s", jobID, record.Details, status)
 						}
 					}
 				}
 
-				jobIdsClause := fmt.Sprint(strings.Join(jobIDStrings, ", "))
+				jobIdsClause := strings.Join(jobIDStrings, ", ")
+				defer func() {
+					// Clear the system.jobs table for the next test run.
+					th.sqlDB.Exec(t, fmt.Sprintf("DELETE FROM system.jobs WHERE id IN (%s)", jobIdsClause))
+					th.sqlDB.Exec(t, fmt.Sprintf("DELETE FROM system.job_info WHERE job_id IN (%s)", jobIdsClause))
+				}()
 
-				// Execute the command and verify its executed on the expected number of rows
+				// Execute the command and verify it is executed on the expected number of rows.
 				numEffected, err := th.cfg.DB.Executor().ExecEx(
 					context.Background(),
 					"test-num-effected",
 					nil,
 					sessiondata.RootUserSessionDataOverride,
 					commandQuery,
 				)
-				require.NoError(t, err)
+				require.NoError(t, err, "%s", delayedShowJobs(t, subJobs))
 
-				// Jobs in the starting state should be affected
+				// Jobs in the starting state should be affected.
 				numExpectedJobsAffected := numJobsPerStatus * len(tc.startingStates)
-				require.Equal(t, numExpectedJobsAffected, numEffected)
+				require.Equal(t, numExpectedJobsAffected, numEffected+1, "%s", delayedShowJobs(t, subJobs))
```

Comment:
was this +1 to force a failure and test the additional debugging?

Reply:
yes, sorry it was indeed to test the additional debugging. I'll remove it before this can merge.